src/Aquifer.API/Endpoints/Resources/Content/ResourceStatusHelpers.cs

```diff
@@ -0,0 +1,127 @@
+        }
+    }
+
+    public static void SetAssignedReviewerUserId(int? assignedReviewerUserId, ResourceContentVersionEntity draftVersion)
+    {
+        if (assignedReviewerUserId is not null)
+        {
+            draftVersion.AssignedReviewerUserId = assignedReviewerUserId;
+        }
+    }
+
+    public static void ValidateAllowedToAssign<TRequest>(AssignmentPermissions permissions,
+        ResourceContentVersionEntity draftVersion,
+        UserEntity user)
+    {
+        var currentUserIsAssigned = draftVersion.AssignedUserId == user.Id;
+        var assignedUserIsInCompany = draftVersion.AssignedUser?.CompanyId == user.CompanyId;
+
+        var allowedToAssign = (permissions.HasAssignOverridePermission &&
+                              (assignedUserIsInCompany || permissions.HasAssignOutsideCompanyPermission)) ||
+                              currentUserIsAssigned;
+
+        if (!allowedToAssign)
+        {
+            ValidationContext<TRequest>.Instance.ThrowError($"Unable to assign user for id {draftVersion.ResourceContentId}");
+        }
+    }
+
+    public static async Task<List<ResourceContentVersionEntity>> GetDraftVersions<TRequest>(int[] contentIds, AquiferDbContext dbContext,
+        CancellationToken ct)
+    {
+        var draftVersions = await dbContext.ResourceContentVersions
+            .AsTracking()
+            .Where(rcv => contentIds.Contains(rcv.ResourceContentId) && rcv.IsDraft).Include(rcv => rcv.ResourceContent)
+            .Include(rcv => rcv.AssignedUser).Include(rcv => rcv.ResourceContentVersionSnapshots)
+            .ToListAsync(ct);
+
+        if (draftVersions.Count != contentIds.Length)
+        {
+            ValidationContext<TRequest>.Instance.ThrowError("One or more resources not found or not in draft status");
+        }
+
+        return draftVersions;
+    }
+
+    public static async Task ValidateReviewerAndAssignedUser<TRequest>(int assignedUserId, int? assignedReviewerUserId,
+        AquiferDbContext dbContext, UserEntity user, AssignmentPermissions permissions,
+        CancellationToken ct)
+    {
+        var userToAssign = await dbContext.Users
+            .AsTracking()
+            .SingleOrDefaultAsync(u => u.Id == assignedUserId && u.Enabled, ct);
+        if (userToAssign is null)
+        {
+            ValidationContext<TRequest>.Instance.ThrowError(Helpers.InvalidUserIdResponse);
+        }
+
+        if (userToAssign.CompanyId != user.CompanyId && !permissions.HasAssignOutsideCompanyPermission)
+        {
+            ValidationContext<TRequest>.Instance.ThrowError("Unable to assign to a user outside your company");
+        }
+
+        if (assignedReviewerUserId is not null)
+        {
+            var reviewer = await dbContext.Users
+                .Where(u => u.Id == assignedReviewerUserId
+                            && u.Enabled
+                            && u.CompanyId == userToAssign!.CompanyId
```

Comment:
Can remove the bang operator here since already verified `userToAssign` isn't null.